### PR TITLE
Bluetooth: controller: Fix channel selection

### LINKED ISSUE #55736: Bluetooth: Controller: Undefined channel selection behavior in lll_chan_iso_subevent method
**Describe the bug**
The _lll_chan_iso_subevent_ method in https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/controller/ll_sw/lll_chan.c contains a bug in relation to the specification.

The next subevent index is calculated via the following line of code:
`chan_idx = ((((uint32_t)prn_subevent_se * x) >> 16) + d + *remap_idx) % chan_count;`

Based on the specification, the outcome of this formula should be treated as a remapping index (since it is already based on the remapping index of the previous subevent). But the code treats it as a channel index that, similarly to the first subevent calculation in _lll_chan_iso_event_, needs to be checked against the channel map and remapped if necessary... The following if-else clause is executed, followed by the return:
```
if ((chan_map[chan_idx >> 3] & (1 << (chan_idx % 8))) == 0U) {
	*remap_idx = chan_idx;
	chan_idx = chan_sel_remap(chan_map, *remap_idx);
} else {
	*remap_idx = chan_idx;
}

return chan_idx;
```

This means that the actual expected behavior (outcome of formula used as remapping index) is only performed in case the outcome of the formula represents an index that is not part of the channel map. Otherwise (else clause), the outcome is treated as channel index and remapping index is only just set to that index as well... This is not detected in the tests, because of the following reasons:

- Test cases using channel map of 37 channels: In this case, remapping will never be done so no outcome of the formula will not be part of the channel map and the buggy else-clause will thus always execute. However, this will not be an issue here since the remapping index **will always represent the actual channel index here.**
- Test cases using channel map [0x00, 0x06, 0xE0, 0x00, 0x1E]: All the cases here lead to remapping indices <= 8 and no channels <= 8 are part of the channel map so the "non-buggy" if-clause above will always be executed.

**To Reproduce**

For instance, when testing out the code using the following parameters:

```
uint8_t chan_map[] = {11, 42, 19, 217, 25};
uint8_t const chan_map_count= 17U;
uint16_t const chan_id = 0x305F;
uint8_t m;
uint16_t remap_idx;
uint16_t prn_s;

/* BIS subevent 1, event counter 8 */
m = lll_chan_iso_event(8, chan_id, chan_map, chan_map_count, &prn_s, &remap_idx);

/* BIS subvent 2 */
m = lll_chan_iso_subevent(chan_id, chan_map, chan_map_count, &prn_s, &remap_idx);

/* BIS subvent 3 */
m = lll_chan_iso_subevent(chan_id, chan_map, chan_map_count, &prn_s, &remap_idx);

/* BIS subvent 4 */
m = lll_chan_iso_subevent(chan_id, chan_map, chan_map_count, &prn_s, &remap_idx);
```

The outcome for channel indices and remapping index will be 

- First subevent channel 17 (remapping index 7)
- Second subevent channel 11 (remapping index 11)
- Third subevent channel 3 (remapping index 3)
- Fourth subevent channel 13 (remapping index 13)

This is undefined behavior in relation to the specification.

**Expected behavior**

The expected outcome of the previous example should be:

- First subevent channel 17 (remapping index 7)
- Second subevent channel 28 (remapping index 11)
- Third subevent channel 9 (remapping index 3)
- Fourth subevent channel 31 (remapping index 13)

This can be achieved by replacing the if-else clause and subsequent return line, with the following code:

```
*remap_idx = chan_idx;
return chan_sel_remap(chan_map, *remap_idx);
```

**Impact**
The channels in the example with if-else clause will never be not part of the channel map, so the bug is not blacklist-breaking. However, the current behavior is not what you would expect in relation to the specification and is thus a major showstopper towards defined behavior when using a channel map for blacklisting on an Isochronous Stream. 

**Environment (please complete the following information):**
 - Commit SHA: 53db4a0b1d16d5e3554e49bb21567ad3226c9ea0 (most recent commit on the file containing the buggy _lll_chan_iso_subevent_ method)

### PR DESCRIPTION
**Pull request by heml-ot**
For ISO-subevents channel selection must use remap table always -regardless of channel is a mapped or unmapped channel.

Fixes #55736.

Signed-off-by: Henrik Møller <heml@demant.com>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/lll_chan.c [hunk 1]
```diff
@@ -147,15 +147,9 @@ uint8_t lll_chan_iso_subevent(uint16_t chan_id, uint8_t *chan_map,
...
 
-	chan_idx = ((((uint32_t)prn_subevent_se * x) >> 16) +
-		    d + *remap_idx) % chan_count;
-
-	if ((chan_map[chan_idx >> 3] & (1 << (chan_idx % 8))) == 0U) {
-		*remap_idx = chan_idx;
-		chan_idx = chan_sel_remap(chan_map, *remap_idx);
-	} else {
-		*remap_idx = chan_idx;
-	}
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/lll_chan.c:158]: This has been tested by `lll_chan_sel_2_ut()` with the sample data in the specification. Please add a testcase that fails with the implementation.

**Review by cvinayak — DISMISSED** [review]

File: subsys/bluetooth/controller/ll_sw/lll_chan.c (hunk 1 above)
Comment (heml-ot) [reply subsys/bluetooth/controller/ll_sw/lll_chan.c:158]: Yes, I know it’s tested against reference data from spec. Unfortunately, the sample data in spec. is poorly chosen. For a mapped ch; as long as the ch_idx and the remap_idx are identical the issue will not reveal itself. We need a different ch map to trigger the issue.
Point taken; I’ll provide a test catching the issue.

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/lll_chan.c:155]: I have added 2 used channels channel map to cover the change.

**Review by cvinayak — DISMISSED** [review]

**Comment by cvinayak** [comment]
@heml-ot and @wopu-ot can someone from your end approve this?

**Review by heml-ot — COMMENTED** [review]
yes, this will work for me.

**Review by cvinayak — APPROVED** [review]

**Review by wopu-ot — APPROVED** [review]
LGTM

**Pull request merged**